crates/apub_objects/src/objects/post.rs

```diff
@@ -302,6 +326,21 @@ impl Object for ApubPost {
     let timestamp = page.updated.or(page.published).unwrap_or_else(Utc::now);
     let post = Post::insert_apub(&mut context.pool(), timestamp, &form).await?;
     plugin_hook_after("after_receive_federated_post", &post)?;
+
+    // write post tags
+    let post_tag_ap_ids = page
+      .tag
+      .iter()
+      .filter_map(HashtagOrLemmyTag::community_tag_url)
+      .collect::<HashSet<_>>();
+    let community_tags = Tag::read_for_community(&mut context.pool(), post.community_id).await?;
+    let post_tags = community_tags
+      .into_iter()
+      .filter(|t| post_tag_ap_ids.contains(&t.ap_id))
+      .map(|t| t.id)
+      .collect();
+    update_post_tags(&post, &post_tags, context).await?;
```

Comment:
I know the update_post_tags will throw an error if that tag isn't in the superset of community tags. I'm thinking about newly created community tags... but I guess that won't be an issue since the federate server won't see them until that community and its tags get refetched anyway.